### PR TITLE
JUnit codegen

### LINKED ISSUE #1039: [Feature] add junit to playwright codegen
similar to https://github.com/microsoft/playwright-dotnet/issues/2247

### PR DESCRIPTION
**Pull request by jfgreffier**
Codgen for Java JUnit

Fixes https://github.com/microsoft/playwright-java/issues/1039

### CODE REVIEW

File: packages/playwright-core/src/server/recorder/java.ts [hunk 1]
```diff
@@ -136,6 +153,46 @@ export class JavaLanguageGenerator implements LanguageGenerator {
...
   generateHeader(options: LanguageGeneratorOptions): string {
     const formatter = new JavaScriptFormatter();
+    if (this._mode === 'junit') {
+      formatter.add(`
+      import com.microsoft.playwright.*;
+      import com.microsoft.playwright.options.*;
+      import org.junit.jupiter.api.*;
+      import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;
+
+      public class TestExample {
```
Comment (yury-s) [inline packages/playwright-core/src/server/recorder/java.ts:163]: I like the idea of generating a complete test, but having all boilerplate code in the generated snippet can be a bit much. Let's introduce base classes for junit tests inside `playwright-java` repo, and have codegen use them? You can see how it is done for .Net. Initially JUnit classes can provide just the basic support and we should be able to evolve them over time by adding things like automatic trace collection etc.

Comment (jfgreffier) [reply packages/playwright-core/src/server/recorder/java.ts:163]: PR here https://github.com/microsoft/playwright-java/pull/1366

**Comment by jfgreffier** [comment]
To be rework after [Feature] JUnit5 integration https://github.com/microsoft/playwright-java/issues/1369

**Comment by pavelfeldman** [comment]
Closing it for now as we are exploring a better JUnit harness support that would be prerequisite to this.

**Pull request closed without merging**